### PR TITLE
Update to use new display-url-api for user facing page links

### PR DESCRIPTION
**Pull request by i386**
This allows Blue Ocean to specify what links get sent in HipChat messages.

@jenkinsci/code-reviewers

### CODE REVIEW

**Comment by imod** [comment]
👍

File: src/test/java/jenkins/plugins/hipchat/utils/BuildUtilsTest.java [hunk 1]
```diff
@@ -129,12 +133,12 @@ public void collectedParametersContainDurationFromBuild() throws Exception {
     public void collectedParametersContainCorrectUrl() throws Exception {
         setupMocks();
 
-        given(jenkins.getRootUrl()).willReturn("http://localhost:8080/jenkins");
-        given(run.getUrl()).willReturn("/job/test%20project/1/");
+        given(jenkins.getRootUrl()).willReturn("http://localhost:" + rule.getLocalPort() + "/jenkins/");
+        given(run.getUrl()).willReturn("job/test%20project/1/");
```
Comment (batmat) [inline src/test/java/jenkins/plugins/hipchat/utils/BuildUtilsTest.java]: Not sure I get why slashes are moving, but well.

**Comment by batmat** [comment]
:+1:

**Comment by i386** [comment]
getURL never returns a / prefix

**Comment by i386** [comment]
@aldaris what do you think?

File: src/test/java/jenkins/plugins/hipchat/utils/BuildUtilsTest.java (hunk 1 above)
Comment (aldaris) [inline src/test/java/jenkins/plugins/hipchat/utils/BuildUtilsTest.java]: This looks correct as far as I can tell, the getRootURL method in Jenkins calls Util.ensureEndsWith(url,"/"); ,so most likely the test was incorrectly set up previously.

**Comment by aldaris** [comment]
If I may suggest one change:

```
diff --git a/src/test/java/jenkins/plugins/hipchat/utils/BuildUtilsTest.java b/src/test/java/jenkins/plugins/hipchat/utils/BuildUtilsTest.java
index 10c1988..c5c777c 100644
--- a/src/test/java/jenkins/plugins/hipchat/utils/BuildUtilsTest.java
+++ b/src/test/java/jenkins/plugins/hipchat/utils/BuildUtilsTest.java
@@ -32,7 +32,7 @@ import org.mockito.runners.MockitoJUnitRunner;
 public class BuildUtilsTest {
 
     @Rule
-    public JenkinsRuleWithServerPort rule = new JenkinsRuleWithServerPort();
+    public JenkinsRule rule = new JenkinsRule();
     @Mock
     private Jenkins jenkins;
     @Mock
@@ -133,12 +133,12 @@ public class BuildUtilsTest {
     public void collectedParametersContainCorrectUrl() throws Exception {
         setupMocks();
 
-        given(jenkins.getRootUrl()).willReturn("http://localhost:" + rule.getLocalPort() + "/jenkins/");
+        given(jenkins.getRootUrl()).willReturn(getRootUrl());
         given(run.getUrl()).willReturn("job/test%20project/1/");
 
         Map<String, String> collected = buildUtils.collectParametersFor(jenkins, run);
 
-        assertThat(collected).containsEntry("URL", "http://localhost:" + rule.getLocalPort() + "/jenkins/job/test%20project/1/");
+        assertThat(collected).containsEntry("URL", getRootUrl() + "job/test%20project/1/");
     }
 
     @Test
@@ -255,9 +255,7 @@ public class BuildUtilsTest {
         }
     }
 
-    final class JenkinsRuleWithServerPort extends JenkinsRule {
-        public int getLocalPort() {
-            return localPort;
-        }
+    private String getRootUrl() throws Exception {
+        return rule.getURL().toString();
     }
 }
```

**Comment by aldaris** [comment]
One final thing: could you squash the commits please? Thanks!

**Comment by i386** [comment]
@aldaris patch applied and all commits squashed :)

**Comment by i386** [comment]
Thanks mate!

**Comment by rickthepick** [comment]
Hi.  One of our Jenkins users uses Blue Ocean - the others don't.  I noticed that *all* the links in HipChat messages now link to Blue Ocean pages.  How can I turn that off?

**Comment by aldaris** [comment]
I don't believe that's possible currently.

**Comment by i386** [comment]
I'll release some updates today that will make it possible to disable on a per user basis

**Pull request merged**